libraries/chain/kv_context.cpp

```diff
@@ -6,22 +6,22 @@
 namespace eosio { namespace chain {
 
    namespace {
-      void kv_resource_manager_update_ram(apply_context& context, int64_t delta, const kv_resource_trace& trace) {
+      void kv_resource_manager_update_ram(apply_context& context, int64_t delta, const kv_resource_trace& trace, account_name payer) {
          std::string event_id;
          if (context.control.get_deep_mind_logger() != nullptr) {
             event_id = STORAGE_EVENT_ID("${id}", ("id", fc::to_hex(trace.key.data(), trace.key.size())));
          }
 
-         context.add_ram_usage(account_name(context.get_receiver()), delta, storage_usage_trace(context.get_action_id(), event_id.c_str(), "kv", trace.op_to_string()));
+         context.add_ram_usage(payer, delta, storage_usage_trace(context.get_action_id(), event_id.c_str(), "kv", trace.op_to_string()));
       }
 
-      void kv_resource_manager_update_disk(apply_context& context, int64_t delta, const kv_resource_trace& trace) {
+      void kv_resource_manager_update_disk(apply_context& context, int64_t delta, const kv_resource_trace& trace, account_name payer) {
          std::string event_id;
          if (context.control.get_deep_mind_logger() != nullptr) {
             event_id = STORAGE_EVENT_ID("${id}", ("id", fc::to_hex(trace.key.data(), trace.key.size())));
          }
 
-         context.add_disk_usage(account_name(context.get_receiver()), delta, storage_usage_trace(context.get_action_id(), event_id.c_str(), "kv", trace.op_to_string()));
+         context.add_disk_usage(payer, delta, storage_usage_trace(context.get_action_id(), event_id.c_str(), "kv", trace.op_to_string()));
```

Comment:
Same as with `update_ram`.  Also, need to forbid increasing disk usage in a notify.